provider/resource_opensearch_index_template.go

```diff
@@ -54,18 +53,11 @@ func resourceOpensearchIndexTemplateRead(d *schema.ResourceData, meta interface{
 
 	var result string
 	var err error
-	esClient, err := getClient(meta.(*ProviderConf))
+	osClient, err := getClient(meta.(*ProviderConf))
 	if err != nil {
 		return err
 	}
-	switch client := esClient.(type) {
-	case *elastic7.Client:
-		result, err = elastic7IndexGetTemplate(client, id)
-	case *elastic6.Client:
-		result, err = elastic6IndexGetTemplate(client, id)
-	default:
-		return errors.New("opensearch version not supported")
-	}
+	result, err = elastic7IndexGetTemplate(osClient, id)
```

Comment:
This introduces a nasty bug! `opensearch_index_template` is supposed to call `_template` not `_index_template`.   Now this doesn't behave as documented.  The documentation states that `opensearch_composable_index_template` calls `_index_template` and you should use `opensearch_index_template` on `opensearch <2.0` but now both resources use the same endpoint.